modules/basics/src/main/java/com/opengamma/strata/basics/currency/LazyFxRateProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.currency;
+
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * An {@code FxRateProvider} that delays fetching its underlying provider
+ * until actually necessary.
+ * <p>
+ * This is typically useful where you <em>may</em> need a {@code MarketDataFxRateProvider}
+ * but want to delay loading market data to construct the provider until you are sure you actually do need it.
+ */
+public class LazyFxRateProvider implements FxRateProvider {
+
+  private final Supplier<FxRateProvider> target;
+
+  private LazyFxRateProvider(Supplier<FxRateProvider> target) {
+    ArgChecker.notNull(target, "target");
+    this.target = Suppliers.memoize(target::get);
+  }
+
+  /**
+   * Create a new LazyFxRateProvider.
+   *
+   * @param fxRateProviderSupplier a supplier of a rate provider
+   * @return the rate provider
+   */
+  public static LazyFxRateProvider of(Supplier<FxRateProvider> fxRateProviderSupplier) {
```

Comment:
Return the interface, not the concrete type

Reply:
What's the reason for that? It doesn't seem consistent with the factories of other `FxRateProvider` implementations.

Reply:
If the type returned is a bean, it should be returned. Otherwise, there is no need to refer to the type. In fact, looking at it again, it probably makes sense to make the new class package scoped and put the static factory method on the interface. We've done this a lot in Strata to minimise the exposed API.